mount/mount_linux.go

```diff
@@ -117,9 +170,44 @@ func (m *Mount) mount(target string) (err error) {
 		// Remount the bind to apply read only.
 		return unix.Mount("", target, "", uintptr(oflags|unix.MS_REMOUNT), "")
 	}
+
+	// remap non-overlay mount point
+	if opt.uidmap != "" && opt.gidmap != "" && m.Type != "overlay" {
+		if err := IDMapMount(target, target, usernsFd); err != nil {
```

Comment:
ping @AkihiroSuda based on your comment https://github.com/containerd/containerd/pull/5890#issuecomment-1586935576, I think we should focus on supporting overlay idmapped mount, if my understanding is correct.

IMO, we should focus on overlayfs idmapped mount at first.

Reply:
This will only be executed if we have no/one parent(s) (we use `bind` mount), so we need a way to remap it (your comment https://github.com/containerd/containerd/pull/5890#issuecomment-1501900391) or we can use the old `chown` approach.

Check this code https://github.com/artqzn/containerd/blob/05b34a00caa401d3f8db0ecf5f39f268124f93dd/snapshots/overlay/overlay.go#L535. I don't see any other case for this, do you?

Reply:
Oh. Good catch! Yes. we should handle the only one layer. Thank you!